File: Back/Events/src/main/java/com/example/supplier/Entity/Event.java
package com.example.supplier.Entity;


import jakarta.persistence.*;
import jakarta.ws.rs.DefaultValue;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
@Builder
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long Id ;

    private EventType Type;
    private String Name;
    // event will be at
    private Date Date;
    // the event created in database
    private Date Created_at;
    // reference USER
    private long Created_by;
    private String Note;
    // a number
    private long ViewsCounter;
    // a number
    private long InterestedCounter;
    // active is before and during the event date ------ will check it
    private boolean Active;

    private Date DisableDate;
    //  a rating table to calculate it
    private double Rating;
    // cant remove this , it is a checkbox that the event creator MUST select or NOT
    @Column(nullable = false, columnDefinition = "boolean default false")
    @DefaultValue("false")
    private boolean Collaboration;
    private Date LastModified_at;
    private long LastModified_by;
    private String DepositNotes;



    /// One to Many    UNIDIRECTIONNEL -----> i dont have accss to the USERs
    @OneToMany(cascade = CascadeType.ALL)
    private Set<InterestedBy> InterestedBys;

    /// One to Many    UNIDIRECTIONNEL   -- ----  association de collaboration
    // les personne qui peut voir les stats du event/admin
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Collaboration> Collaborations;

    /// One to Many    UNIDIRECTIONNEL
    // la liste des commentaire / ratings
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Rating> Ratings;


}

File: Front/projects/back-office/src/app/models/event.ts
// event.ts

import { Image } from "./image";

export class Event {
    id!: number;
    type!: EventType; // Assuming EventType is also transformed to TypeScript
    name!: string;
    date!: Date;
    createdAt!: Date;
    createdBy!: number;
    note!: string;
    viewsCounter!: number;
    interestedCounter!: number;
    active!: boolean;
    rating!: number;
    collaboration!: boolean;
    lastModifiedAt!: Date;
    lastModifiedBy!: number;
    depositNotes!: string;

      imageUrl?:string
     imageId?:string
  
    interestedBys?: InterestedBy[]; // Assuming InterestedBy is another TypeScript interface
    images?: Image[];
    ratings?: Rating[];
  }
  
  // Assuming the following interfaces for related entities
  
  export class InterestedBy {
    id!: number;
    userID!: number;
    eventID!: number;
    timestamp!: Date;
  }

  export class Rating {
    id!: number;
    userID!: number;
    eventID!: number;
    rating!: number;
    comment!: string;
  }
  
// event-type.enum.ts

export enum EventType {
    Enterprise_Day = "Enterprise Day",
    Club_Day = "Club Day",
    Association_Day = "Association Day",
  }
  